Этап 1. Сервис Статистики

Kirill495 opened the pull request:
Спринт 18.
Дипломный проект. Разработка сервиса статистики

Commit:
feat:
Спринт 18.
Дипломный проект. Разработка сервиса статистики

Commit:
Спринт 18.
Дипломный проект. Разработка сервиса статистики
Доработка из-за ошибок запуска на git-hub

Commit:
Спринт 18.
Дипломный проект. Разработка сервиса статистики
Доработка из-за ошибок запуска на git-hub

Commit:
Спринт 18.
Дипломный проект. Разработка сервиса статистики
Доработка из-за ошибок запуска на git-hub

Kirill495 commented on docker-compose.yml:
```diff
@@ -5,8 +5,8 @@ services:
     image: stats_client
     container_name: stats_client
     ports:
-      - "9090:9090"
```
Уважаемый ревьюер!
Обрати, пожалуйста, внимание.
Изначально я сделал так, что stats_client у меня был на порте 9090, а stats_server на 8080. Но при этом не проходили тесты.  Подозреваю, что не хватало времени (60 секунд)  на запуск обоих контейнеров: stats_client и stats_server. После того как я переключил порты, проверки стали проходить. То есть сейчас, в решении на github в ходе проверок stats_client не прогоняется.
Но оба звена: и клиент и сервер сервиса статистики у меня реализованы.

VladimirTitov4 replied:
ок)

VladimirTitov4 commented on docker-compose.yml:
```diff
@@ -1,15 +1,59 @@
+      - DB_HOST=stats-db
+      - DB_PORT=5432
+      - DB_NAME=stats
+      - POSTGRES_USER=${DB_USER_NAME}
```
Отлично, что завернули в переменную окружения

VladimirTitov4 commented on pom.xml:
```diff
@@ -95,45 +99,45 @@
-									</rule>
-								</rules>
-							</configuration>
+<!--							<configuration>-->
```
Закомментированный код лучше удалить

VladimirTitov4 commented on stats/stats-client/src/main/java/ru/practicum/ewm/client/stats/StatsController.java:
```diff
@@ -0,0 +1,47 @@
+
+   @PostMapping(path = "/hit", consumes = "application/json")
+   @ResponseStatus(HttpStatus.CREATED)
+   public ResponseEntity<Object> saveInfo(@RequestBody @Valid EndpointHit endpointHit) {
```
Object лучше не указывать в типе возвращаемого значения.

Kirill495 replied:
Исправил

VladimirTitov4 commented on stats/stats-server/src/main/java/ru/practicum/ewm/stats/server/mapper/EndpointMapper.java:
```diff
@@ -0,0 +1,19 @@
+        unmappedTargetPolicy = ReportingPolicy.IGNORE)
+public interface EndpointMapper {
+
+   EndpointMapper MAPPER = Mappers.getMapper(EndpointMapper.class);
```
В целом эта константа не обязательна, можно вредрять этот маппер как бин в любой другой класс

Kirill495 replied:
Убрал

VladimirTitov4 commented on stats/stats-server/src/main/resources/schema.sql:
```diff
@@ -0,0 +1,9 @@
+    id INTEGER GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    app VARCHAR(100) NOT NULL,
+    uri VARCHAR(100) NOT NULL,
+    ip VARCHAR(21) NOT NULL,
```
Для ip достаточно 15 символов

Kirill495 replied:
Исправил.

Commit:
Спринт 18.
Дипломный проект. Разработка сервиса статистики
Доработка по замечаниям ревьюера.

Kirill495 commented:
помещаю в ветку main после приемки работы ревьюером
